viper.py

```diff
@@ -169,118 +165,116 @@ def SSRstat(vgrid, SSR, dk=1, plot='maybe', N=None):
     argopt('-stepRV', help='step through fixed RVs to find the minimum in the rms (a: (auto) picks the fixed RVs automatically to get close to the minimum; m: (manual) uses fixed range and steps around vguess)', choices=['a', 'm'], type=str)
     argopt('-tag', help='Output tag for filename', default='tmp', type=str)
     argopt('-targ', help='Target name requested in simbad for coordinates, proper motion, parallax and absolute RV.', dest='targname')
-    argopt('-telluric', help='mask: mask telluric; sig: downweigth tellurics; add: telluric forward modelling (just for CRIRES+)', default=None, type=str)
+    argopt('-telluric', help='mask: mask telluric; sig: downweight tellurics; add: telluric forward modelling (just for CRIRES+)', default=None, type=str)
     argopt('-tellshift', help='Variable telluric wavelength shift (one value for all selected molecules)', action='store_true')
     argopt('-tsig', help='(relative) sigma value for weighting tellurics', default=1, type=float)
-    argopt('-vg', help='RV guess', default=1., type=float)   # slightly offsetted
+    argopt('-rv_guess', help='RV guess', default=1., type=float)   # slightly offsetted
     argopt('-?', '-h', '-help', '--help',  help='show this help message and exit', action='help')
 
     
-
     args = parser.parse_args()
     globals().update(vars(args))
 
 
-def fit_chunk(o, chunk, obsname, targ=None, tpltarg=None):
+def fit_chunk(order, chunk, obsname, targ=None, tpltarg=None):
     ####  observation  ####
-    x, w, f, e, bp, bjd, berv = Spectrum(obsname, o=o, targ=targ)
-    i = np.arange(f.size)
+    pixel, wave_obs, spec_obs, err_obs, flag_obs, bjd, berv = Spectrum(obsname, order=order, targ=targ)
 
     if telluric == 'mask':
-        bp[mskatm(w) > 0.1] |= flag.atm
-        bp[np.isnan(f)] |= flag.nan
+        flag_obs[mskatm(wave_obs) > 0.1] |= flag.atm
+        flag_obs[np.isnan(spec_obs)] |= flag.nan
 
-    lmin = max(w[iset][0], w_tpl[o][0], w_I2[0])
-    lmax = min(w[iset][-1], w_tpl[o][-1], w_I2[-1])
+    # select common wavelength range 
+    lmin = max(wave_obs[iset][0], wave_tpl[order][0], wave_cell[0])
+    lmax = min(wave_obs[iset][-1], wave_tpl[order][-1], wave_cell[-1])
 
     # trim the observation to a range valid for the model
     vcut = 100   # [km/s]
-    bp[np.log(w) < np.log(lmin)+vcut/c] |= flag.out
-    bp[np.log(w) > np.log(lmax)-vcut/c] |= flag.out
+    flag_obs[np.log(wave_obs) < np.log(lmin)+vcut/c] |= flag.out
+    flag_obs[np.log(wave_obs) > np.log(lmax)-vcut/c] |= flag.out
+
+    # using the supersampled log(wavelength) space with knot index j
+    sj = slice(*np.searchsorted(lnwave_cell_full, np.log([lmin, lmax])))
+    lnwave_cell = lnwave_cell_full[sj]
+    lnspec_cell = lnspec_cell_full[sj]
```

Comment:
`lnspec_cell` -> `spec_cell`

Reply:
No, not this time. `lnspec_cell` was `iod_j` before- so the ln of the cell spectrum. `spec_cell` was `f_I2` before.

Reply:
I don't understand. In our code only wavelength can be logarithmic, but flux/spec never.

Before it was
https://github.com/mzechmeister/viper/blob/41cdd82c96e09124984b7466f396d74404bf18e6/viper.py#L249-L250

`uj` => lnwave, `iod_j` => `spec_cell`

But it seems to collide here
https://github.com/mzechmeister/viper/blob/7243d7182c5104a0108bf9dea391b85af3a70fed/viper.py#L687

There is already a variable `spec_cell`.

The suffix `_j` was dropped, which distinguished between orginal sampling and resampling to knot `j`

Reply:
maybe suffix `_raw` or `_org`

Reply:
Hmm, on the one side adding `_raw` sounds like a good idea. Still, somehow in this case it feels weird to have no suffix to `spec_cell`, as this sounds more like the "original" cell spectrum. Do you know, what I mean? 
But maybe I am just thinking too much about it plus I have no better idea.

Reply:
I see. I guess we are back to a suffix `_j`. I think, this is not too bad (better than e.g. `mod` for model grid).
(`_j` was not used throughout in the old version.)

`lnspec_cell` => **`spec_j_cell`** or `spec_cell_j`

Currently, it compares like
https://github.com/mzechmeister/viper/blob/41cdd82c96e09124984b7466f396d74404bf18e6/viper.py#L283

https://github.com/mzechmeister/viper/blob/a3228159cc2d24b1e86562c45ade5546d8b8685f/viper.py#L277

So in `atmj` the `j` was dropped.

Note, also `lnwave_cell` is a bit misleading. Actually, the variable means the (super)sampled grid under the convolution.
=> `lnwave_j` (previously `uj`)

Reply:
So, should we keep `wave_cell` and `spec_cell` for `w_I2` and `f_I2`? And than use `lnwave_j` and `spec_cell_j` for `uj` and `iod_j`? I am fine with keeping `_j` under these circumstances and it still fits better to the rest of the variable names and gives a better idea of the variable itself.

Reply:
Yeah, then let's do it so.